src/v/cluster/scheduling/constraints.cc

```diff
@@ -133,6 +133,76 @@ distinct_from(const std::vector<model::broker_shard>& current) {
     return hard_constraint_evaluator(std::make_unique<impl>(current));
 }
 
+hard_constraint_evaluator disk_not_overflowed_by_partition(
+  const cluster_health_report& health_report,
+  const float& max_disc_usage,
+  const size_t& partition_size,
+  const std::unordered_map<model::node_id, uint64_t>&
+    assigned_reallocation_sizes) {
+    class impl : public hard_constraint_evaluator::impl {
+    public:
+        impl(
+          const cluster_health_report& health_report,
+          const float& max_disc_usage,
+          const size_t& partition_size,
+          const std::unordered_map<model::node_id, uint64_t>&
+            assigned_reallocation_sizes)
+          : _health_report(health_report)
+          , _max_disc_usage(max_disc_usage)
+          , _partition_size(partition_size)
+          , _assigned_reallocation_sizes(assigned_reallocation_sizes) {}
+
+        bool evaluate(const allocation_node& node) const final {
+            // we return 0 for fully filled node and 10'000'000 for nodes
+            // with disk capacity available
+            auto equal_node_id =
+              [&node](const node_health_report& health_report) {
+                  return health_report.id == node.id();
+              };
+            auto node_health = std::find_if(
+              _health_report.node_reports.begin(),
+              _health_report.node_reports.end(),
+              equal_node_id);
+            if (node_health == _health_report.node_reports.end()) {
+                return false;
+            } else {
+                // TODO Support multiple disks
+                uint64_t assigned_reallocation_size = 0;
+                if (_assigned_reallocation_sizes.contains(node.id())) {
```

Comment:
nit: can use find here to perform a single lookup instead of two

Reply:
fixed